pkg/config/config_parser.go

```diff
@@ -45,15 +45,20 @@ var configSchemaHCL = &hcl.BodySchema{
 }
 
 func (p *Parser) LoadConfigFromSource(name string, data []byte) (*Config, diag.Diagnostics) {
+	var config *Config
+	var diags diag.Diagnostics
 	if IsNameYAML(name) {
-		return decodeConfigYAML(bytes.NewBuffer(data))
+		config, diags = decodeConfigYAML(bytes.NewBuffer(data))
+	} else {
+		config, diags = p.decodeConfigHCL(name, data)
 	}
 
-	body, diags := p.LoadFromSource(name, data)
```

Comment:
Inlined `LoadFromSource` and handling its result into `decodeConfigHCL`.

Also not sure where `body` can be `nil` as `LoadFromSource` always returns a non `nil` value